### PR TITLE
Compression

### PR DESCRIPTION
**Pull request by jleiandy**
Added optimizations, new server endpoints, relocated static image assets, and refactored thumbnail inputs into divs. Please see commit comments for more details.

### CODE REVIEW

File: client/components/ProductOverview/StyleSelector.jsx [hunk 1]
```diff
@@ -16,9 +16,13 @@ const StyleSelector = ({ styles, currentStyle, updateCurrentStyle }) => {
         <div className="stylesAvailable">
           {styles.map((style, index) => (
             <span className="stylesAvailable__list-item" key={style.style_id}>
-              <input
-                type="image"
-                src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/89/HD_transparent_picture.png/1200px-HD_transparent_picture.png"
+              <div
+                role="button"
+                onKeyPress={(e) => {
```
Comment (cschucode) [inline client/components/ProductOverview/StyleSelector.jsx]: This works for now, but ideally it should be in a handler

File: client/components/ProductOverview/ThumbnailView.jsx [hunk 2]
```diff
@@ -57,14 +57,18 @@ const ThumbnailView = ({ photos, handleClick, currentIndex }) => {
...
+                tabIndex={0}
                 alt="image from selected style set"
                 onClick={(e) => handleClick(e, index)}
+                onKeyPress={(e) => {
+                  e.stopPropagation();
+                  if (e.key === 'Enter') { handleClick(e, index); }
+                }}
                 id={index === currentIndex ? 'selected' : 'unselected'}
                 key={photo.thumbnail_url}
-                src="https://upload.wikimedia.org/wikipedia/commons/thumb/8/89/HD_transparent_picture.png/1200px-HD_transparent_picture.png"
```
Comment (cschucode) [inline client/components/ProductOverview/ThumbnailView.jsx:67]: ❤️

File: package.json [hunk 3]
```diff
@@ -62,6 +63,7 @@
     "react-icons": "^4.2.0",
     "react-moment": "^1.1.1",
     "style-loader": "^3.2.1",
-    "styled-components": "^5.3.1"
+    "styled-components": "^5.3.1",
+    "zlib": "^1.0.5"
```
Comment (cschucode) [inline package.json]: I was under the impression that 'zlib' was included with Node in versions after 10.16. (https://nodejs.org/api/zlib.html)  Do we need to import this?

File: server/index.js [hunk 4]
```diff
@@ -1,20 +1,58 @@
 const express = require('express');
 const path = require('path');
 const axios = require('axios');
+const { join } = require('path');
```
Comment (cschucode) [inline server/index.js:2]: It looks like you import path on line 2. Any reason just to restructure join here? Or, do we not need the path import anymore?

File: client/components/ProductOverview/StyleSelector.jsx (hunk 1 above)
Comment (jleiandy) [reply client/components/ProductOverview/StyleSelector.jsx]: created handler

File: server/index.js (hunk 4 above)
Comment (jleiandy) [reply server/index.js:2]: was just sloppy. refactored.

File: package.json (hunk 3 above)
Comment (jleiandy) [reply package.json]: good catch, i uninstalled the zlib node module and our webpack is still working.

File: client/css/ImageGallery.scss [hunk 5]
```diff
@@ -26,8 +28,7 @@
 .focusedImageExpand {
   background-repeat: no-repeat;
   background-size: 100%;
-  height: 100%;
-  width: 100%;
+  animation: flexanimation 5s ease-out;
```
Comment (cschucode) [inline client/css/ImageGallery.scss:31]: I got to learn about this.

**Review by cschucode — APPROVED** [review]
Let's delete the webpack generated files and add them to the gitignore.

**Pull request merged**